src/EduManagementLab.EfRepository/Repositories/CourseRepository.cs

```diff
@@ -11,6 +11,27 @@ public CourseRepository(DataContext context) : base(context)
         {
             _context = context;
         }
+
+        public IEnumerable<Course.Membership>? GetUserCourses(Guid userId)
+        {
```

Comment:
You can remove the ?
This will never be null. If the cannot find any memberships it will return an empty collection.